### PR TITLE
chorus predict

### PR DESCRIPTION
**Pull request by EBSmash**
hope my bad code helps

### CODE REVIEW

File: src/main/java/cope/cosmos/client/features/modules/misc/ChorusPredict.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    public void onRender3D() {
+        if (tpPos != null) {
+            RenderUtil.drawBox(new RenderBuilder().position(tpPos).color(renderColor).box(Box.FILL).setup().line(1.5F).depth(true).blend().texture());
+
+        }
+    }
+  
+    public void onUpdate(PacketEvent.Receive event) {
```
Comment (linustouchtips) [inline src/main/java/cope/cosmos/client/features/modules/misc/ChorusPredict.java:51]: This event needs to be subscribed to the EventBus, and also the name should be different from the onUpdate method

File: src/main/java/cope/cosmos/client/features/modules/misc/ChorusPredict.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        if (event.getPacket() instanceof SPacketSoundEffect) {
+            if (((SPacketSoundEffect) event.getPacket()).getSound() == SoundEvents.ITEM_CHORUS_FRUIT_TELEPORT) {
```
Comment (linustouchtips) [inline src/main/java/cope/cosmos/client/features/modules/misc/ChorusPredict.java:53]: Use .equals() when comparing objects

File: src/main/java/cope/cosmos/client/features/modules/misc/ChorusPredict.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        for (Entity entity : mc.world.loadedEntityList) {
+            if (entity.getDistance(tpPos.getX(), tpPos.getY(), tpPos.getZ()) < 1) {
+                tpPos = null;
+                break;
+            }
+        }
+
+    }
+
+    public void onRender3D() {
```
Comment (linustouchtips) [inline src/main/java/cope/cosmos/client/features/modules/misc/ChorusPredict.java:43]: Needs an @Override

File: src/main/java/cope/cosmos/client/features/modules/misc/ChorusPredict.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  public ChorusPredict() {
+        super("ChorusPredict", Category.MISC, "Predicts Where Someone will teleport when eating chorus fruit");
+        INSTANCE = this;
+        setExempt(true);
+    }
+
+    public static Setting<Color> renderColor = new Setting<>("TP Color", "Color of the Position they will tp to", new Color(255, 0, 0, 45)).setParent(colors);
+
+
+    public void onUpdate() {
```
Comment (linustouchtips) [inline src/main/java/cope/cosmos/client/features/modules/misc/ChorusPredict.java:32]: Needs an @Override

File: src/main/java/cope/cosmos/client/features/modules/misc/ChorusPredict.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        if (tpPos == null) return;
```
Comment (linustouchtips) [inline src/main/java/cope/cosmos/client/features/modules/misc/ChorusPredict.java:33]: tpPos was never defined

**Comment by linustouchtips** [comment]
Also, this doesn't seem to be predicting the chorus teleport, just rendering it when it happens

**Comment by CynicalMuffin** [comment]
huh whats the point then

**Comment by Sxmurai** [comment]
> Also, this doesn't seem to be predicting the chorus teleport, just rendering it when it happens

thats what chorus predict does lol, the name doesnt make sense but it just checks for a sound packet.

however, a more appropriate name could be ChorusESP

**Pull request closed without merging**